docs/references/ingress-messages.mdx

```diff
@@ -0,0 +1,82 @@
+
+Error handling for queries is typically simple: on error, one can simply retry until a successful response is received. 
+
+## Update calls
+Updates can modify a canister's state and thus go through consensus, typically taking 1-3 seconds. They return certified responses (signed by the subnet), providing authenticity within the subnet's corruption bounds.
+
+Update calls can be [synchronous](/docs/current/references/ic-interface-spec#http-sync-call-overview) or [asynchronous](/docs/current/references/ic-interface-spec#http-async-call-overview). 
+
+A [synchronous `call`](https://internetcomputer.org/docs/current/references/ic-interface-spec#http-call) follows the "call and await" pattern. In the success case, it maintains the HTTPS connection until a certified response is produced, as illustrated below. 
+
+![Dev containers 1](./_attachments/ingress_sync_call.png)
+
+Compared to the [asynchronous `call`](/docs/current/references/ic-interface-spec#http-async-call) endpoint, where the [`read_state`](/docs/current/references/ic-interface-spec#http-read-state) endpoint is used to poll for the request status and obtain the reply, the synchronous endpoint removes the need to poll for the response in the success case. 
+
+However, a response may not be produced until some implementation-specific timeout is reached. In that case, the synchronous endpoint behaves like the asynchronous one: it returns a `202` response, and the client needs to poll [read_state](/docs/current/references/ic-interface-spec#http-read-state) to get the request status and eventually obtain the reply, as illustrated below.
+
+![Dev containers 1](./_attachments/ingress_async_call.png)
+
+### Update call errors and their implications
+
+A number of errors could occur in the above flow. The following are examples of errors, indicated using the numbers in the diagram:
+1. Due to misconfiguration, the agent might fail to successfully sign the update call and not even send it off.
+2. The call to the boundary node may fail, for example because the agent or the boundary node is offline and the call times out.
+3. Calls (`call` or `read_state`) to the IC node may fail, for example due to 
+   1. the node being offline
+   2. an execution pre-processing error (status `200` in case of the async call)
+   3. or the node being incapable of handling the request (status `5xx`)
+4. It could be that status responses are never received. For example, a malicious boundary node could block the status response. Thus, the agent would never see a [status like `received`, `processing`, etc.](/docs/current/references/ic-interface-spec#state-tree-request-status) upon calling `read_state`.
```

Comment:
```suggestion
4. It could be that status responses are never received. For example, the user's might lose network connectivity, or a malicious boundary node could block the status response. Thus, the agent would never see a [status like `received`, `processing`, etc.](/docs/current/references/ic-interface-spec#state-tree-request-status) upon calling `read_state`.
```

Reply:
Thanks, done.